### DISCUSSION TITLE
Printing in seahorn passes

### DISCUSSION THREAD

**Discussion by Nivous** [seahorn/seahorn#534] [General]
I am creating my own pass, and I am trying to print for debug purposes but for some reason cannot see my print. 
Is printing an option? If so how should this be done?
I have done it like this:

`    outs() << "Module:\n";
    M.print(outs(), NULL);`

When before I made sure to use llvm namespace.

**Comment by caballa** [comment]
I think that should work. We do print the module for debugging purposes. See for instance https://github.com/seahorn/seahorn/blob/dev14/lib/seahorn/BvOpSem2.cc#L2441. 

Are you sure that the pass is being called? you can add to the binary `seahorn` options such as `--debug-pass=Structure` to see if your pass has been scheduled (although I think you need to compile LLVM and seahorn in debug mode for that) and you can also pass the option `--print-after-all` to see the bitcode after every LLVM pass.
